### PR TITLE
[CHIP] Add support for Xcode13 beta2.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/chip.cs [hunk 1]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeApplicationSatusWithResponseHandler:")]
+		void ReadAttributeApplicationSatus (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPBarrierControl")]
```
Comment (tj-devel709) [inline src/chip.cs:133]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPBarrierControl")]
```
small

File: src/chip.cs [hunk 2]
```diff
@@ -0,0 +1,1767 @@
...
+	[DisableDefaultCtor]
+	interface ChipBarrierControl
+	{
+
+		[Export ("initWithDevice:endpoint:queue:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
```
Comment (tj-devel709) [inline src/chip.cs:142]: When should we use this ResultTypeName with Async?

File: src/chip.cs [hunk 3]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("writeAttributeLocalConfigDisabledWithValue:responseHandler:")]
+		void WriteAttributeLocalConfigDisabled (byte value, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPBinding")]
```
Comment (tj-devel709) [inline src/chip.cs:270]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPBinding")]
```
same

File: src/chip.cs [hunk 4]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("initWithDevice:endpoint:queue:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	delegate void ChipResponseHandler ([NullAllowed] NSError error, [NullAllowed] NSDictionary data);
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPApplicationBasic")]
```
Comment (tj-devel709) [inline src/chip.cs:91]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPApplicationBasic")]
```

**Review by tj-devel709 — APPROVED** [review]
Other than the missing spaces in the basetypes, it looks good to me!

File: src/chip.cs (hunk 4 above)
Comment (tj-devel709) [reply src/chip.cs:91]: There a few others as well!

File: src/chip.cs (hunk 2 above)
Comment (mandel-macaque) [reply src/chip.cs:142]: It is related when the handler returns more than one value:
```csharp
delegate void ChipResponseHandler ([NullAllowed] NSError error, [NullAllowed] NSDictionary data);
```

The generator will tell you.

File: src/chip.cs [hunk 5]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeBarrierPositionWithResponseHandler:")]
+		void ReadAttributeBarrierPosition (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPBasic")]
```
Comment (mandel-macaque) [inline src/chip.cs:172]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPBasic")]
```

File: src/chip.cs [hunk 6]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("unbind:groupId:endpointId:clusterId:responseHandler:")]
+		void Unbind (ulong nodeId, ushort groupId, byte endpointId, ushort clusterId, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPColorControl")]
```
Comment (mandel-macaque) [inline src/chip.cs:293]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPColorControl")]
```

File: src/chip.cs [hunk 7]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("writeAttributeStartUpColorTemperatureMiredsWithValue:responseHandler:")]
+		void WriteAttributeStartUpColorTemperatureMireds (ushort value, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPDescriptor")]
```
Comment (mandel-macaque) [inline src/chip.cs:656]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPDescriptor")]
```

File: src/chip.cs [hunk 8]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributePartsListWithResponseHandler:")]
+		void ReadAttributePartsList (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPDoorLock")]
```
Comment (mandel-macaque) [inline src/chip.cs:687]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPDoorLock")]
```

File: src/chip.cs [hunk 9]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeActuatorEnabledWithResponseHandler:")]
+		void ReadAttributeActuatorEnabled (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPGeneralCommissioning")]
```
Comment (mandel-macaque) [inline src/chip.cs:814]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPGeneralCommissioning")]
```

File: src/chip.cs [hunk 10]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("writeAttributeBreadcrumbWithValue:responseHandler:")]
+		void WriteAttributeBreadcrumb (ulong value, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPGroupKeyManagement")]
```
Comment (mandel-macaque) [inline src/chip.cs:853]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPGroupKeyManagement")]
```

File: src/chip.cs [hunk 11]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeGroupKeysWithResponseHandler:")]
+		void ReadAttributeGroupKeys (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPGroups")]
```
Comment (mandel-macaque) [inline src/chip.cs:876]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPGroups")]
```

File: src/chip.cs [hunk 12]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeNameSupportWithResponseHandler:")]
+		void ReadAttributeNameSupport (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPIdentify")]
```
Comment (mandel-macaque) [inline src/chip.cs:919]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPIdentify")]
```

File: src/chip.cs [hunk 13]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("writeAttributeIdentifyTimeWithValue:responseHandler:")]
+		void WriteAttributeIdentifyTime (ushort value, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPLevelControl")]
```
Comment (mandel-macaque) [inline src/chip.cs:949]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPLevelControl")]
```

File: src/chip.cs [hunk 14]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("reportAttributeCurrentLevelWithResponseHandler:")]
+		void ReportAttributeCurrentLevel (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPLowPower")]
```
Comment (mandel-macaque) [inline src/chip.cs:1008]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPLowPower")]
```

File: src/chip.cs [hunk 15]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("sleep:")]
+		void Sleep (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPNetworkCommissioning")]
```
Comment (mandel-macaque) [inline src/chip.cs:1027]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPNetworkCommissioning")]
```

File: src/chip.cs [hunk 16]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("updateWiFiNetwork:credentials:breadcrumb:timeoutMs:responseHandler:")]
+		void UpdateWiFiNetwork (NSData ssid, NSData credentials, ulong breadcrumb, uint timeoutMs, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPOnOff")]
```
Comment (mandel-macaque) [inline src/chip.cs:1078]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPOnOff")]
```

File: src/chip.cs [hunk 17]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("reportAttributeOnOffWithResponseHandler:")]
+		void ReportAttributeOnOff (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPOperationalCredentials")]
```
Comment (mandel-macaque) [inline src/chip.cs:1117]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPOperationalCredentials")]
```

File: src/chip.cs [hunk 18]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeFabricsListWithResponseHandler:")]
+		void ReadAttributeFabricsList (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPPumpConfigurationAndControl")]
```
Comment (mandel-macaque) [inline src/chip.cs:1152]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPPumpConfigurationAndControl")]
```

File: src/chip.cs [hunk 19]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("writeAttributeOperationModeWithValue:responseHandler:")]
+		void WriteAttributeOperationMode (byte value, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPScenes")]
```
Comment (mandel-macaque) [inline src/chip.cs:1207]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPScenes")]
```

File: src/chip.cs [hunk 20]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeNameSupportWithResponseHandler:")]
+		void ReadAttributeNameSupport (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPSwitch")]
```
Comment (mandel-macaque) [inline src/chip.cs:1270]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPSwitch")]
```

File: src/chip.cs [hunk 21]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("reportAttributeCurrentPositionWithResponseHandler:")]
+		void ReportAttributeCurrentPosition (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPTemperatureMeasurement")]
```
Comment (mandel-macaque) [inline src/chip.cs:1301]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPTemperatureMeasurement")]
```

File: src/chip.cs [hunk 22]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeMaxMeasuredValueWithResponseHandler:")]
+		void ReadAttributeMaxMeasuredValue (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPTestCluster")]
```
Comment (mandel-macaque) [inline src/chip.cs:1336]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPTestCluster")]
```

File: src/chip.cs [hunk 23]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("readAttributeListInt8uWithResponseHandler:")]
+		void ReadAttributeListInt8u (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPThermostat")]
```
Comment (mandel-macaque) [inline src/chip.cs:1494]:
```suggestion
	[BaseType (typeof (ChipCluster), Name="CHIPThermostat")]
```

File: src/chip.cs [hunk 24]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("writeAttributeSystemModeWithValue:responseHandler:")]
+		void WriteAttributeSystemMode (byte value, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject), Name="CHIPOnboardingPayloadParser")]
```
Comment (mandel-macaque) [inline src/chip.cs:1572]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPOnboardingPayloadParser")]
```

File: src/chip.cs [hunk 25]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("setupPayloadForOnboardingPayload:ofType:error:")]
+		[return: NullAllowed]
+		ChipSetupPayload SetupPayload (string onboardingPayload, ChipOnboardingPayloadType type, [NullAllowed] out NSError error);
+	}
+
+	interface IChipDevicePairingDelegate {}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof(NSObject), Name="CHIPDevicePairingDelegate")]
```
Comment (mandel-macaque) [inline src/chip.cs:1586]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPDevicePairingDelegate")]
```

File: src/chip.cs [hunk 26]
```diff
@@ -0,0 +1,1767 @@
...
+
+		[Export ("onPairingDeleted:")]
+		void OnPairingDeleted (NSError error);
+
+		[Export ("onAddressUpdated:")]
+		void OnAddressUpdated (NSError error);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject), Name="CHIPDeviceController")]
```
Comment (mandel-macaque) [inline src/chip.cs:1603]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPDeviceController")]
```

File: src/chip.cs [hunk 27]
```diff
@@ -0,0 +1,1767 @@
...
+
+		[Export ("startup:")]
+		bool Startup ([NullAllowed] IChipPersistentStorageDelegate storageDelegate);
+
+		[Export ("shutdown")]
+		bool Shutdown();
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject), Name="CHIPError")]
```
Comment (mandel-macaque) [inline src/chip.cs:1656]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPError")]
```

File: src/chip.cs [hunk 28]
```diff
@@ -0,0 +1,1767 @@
...
+		[return: NullAllowed]
+		NSError Create (int errorCode);
+
+		[Static]
+		[Export ("errorToCHIPErrorCode:")]
+		int ToChipErrorCode (NSError errorCode);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject), Name="CHIPManualSetupPayloadParser")]
```
Comment (mandel-macaque) [inline src/chip.cs:1671]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPManualSetupPayloadParser")]
```

File: src/chip.cs [hunk 29]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("populatePayload:")]
+		[return: NullAllowed]
+		ChipSetupPayload PopulatePayload ([NullAllowed] out NSError error);
+	}
+
+	interface IChipPersistentStorageDelegate {}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof(NSObject), Name="CHIPPersistentStorageDelegate")]
```
Comment (mandel-macaque) [inline src/chip.cs:1687]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPPersistentStorageDelegate")]
```

File: src/chip.cs [hunk 30]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("CHIPSetKeyValue:value:")]
+		void SetValue (string key, string value);
+
+		[Abstract]
+		[Export ("CHIPDeleteKeyValue:")]
+		void DeleteValue (string key);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject), Name="CHIPOptionalQRCodeInfo")]
```
Comment (mandel-macaque) [inline src/chip.cs:1704]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPOptionalQRCodeInfo")]
```

File: src/chip.cs [hunk 31]
```diff
@@ -0,0 +1,1767 @@
...
+
+		[Export ("integerValue", ArgumentSemantic.Strong)]
+		NSNumber IntegerValue { get; set; }
+
+		[Export ("stringValue", ArgumentSemantic.Strong)]
+		string StringValue { get; set; }
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject), Name="CHIPSetupPayload")]
```
Comment (mandel-macaque) [inline src/chip.cs:1722]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPSetupPayload")]
```

File: src/chip.cs [hunk 32]
```diff
@@ -0,0 +1,1767 @@
...
+
+		[Export ("serialNumber", ArgumentSemantic.Strong)]
+		string SerialNumber { get; set; }
+
+		[Export ("getAllOptionalVendorData:")]
+		ChipOptionalQRCodeInfo[] GetAllOptionalVendorData ([NullAllowed] out NSError error);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject), Name="CHIPQRCodeSetupPayloadParser")]
```
Comment (mandel-macaque) [inline src/chip.cs:1755]:
```suggestion
	[BaseType (typeof (NSObject), Name="CHIPQRCodeSetupPayloadParser")]
```

File: src/chip.cs [hunk 33]
```diff
@@ -0,0 +1,1767 @@
...
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("setYeardaySchedule:userId:localStartTime:localEndTime:responseHandler:")]
+		void SetYeardaySchedule (byte scheduleId, ushort userId, uint localStartTime, uint localEndTime, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("unlockDoor:responseHandler:")]
+		void UnlockDoor (string pin, ChipResponseHandler responseHandler);
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+
```
Comment (rolfbjarne) [inline src/chip.cs:784]: The [Async] attribute should be one liner lower.

**Review by rolfbjarne — APPROVED** [review]

File: src/chip.cs [hunk 34]
```diff
@@ -0,0 +1,1767 @@
...
+		[Export ("addThreadNetwork:breadcrumb:timeoutMs:responseHandler:")]
+		void AddThreadNetwork (NSData operationalDataset, ulong breadcrumb, uint timeoutMs, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("addWiFiNetwork:credentials:breadcrumb:timeoutMs:responseHandler:")]
+		void AddWiFiNetwork (NSData ssid, NSData credentials, ulong breadcrumb, uint timeoutMs, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("disableNetwork:breadcrumb:timeoutMs:responseHandler:")]
+		void DisableNetwork (NSData networkID, ulong breadcrumb, uint timeoutMs, ChipResponseHandler responseHandler);
```
Comment (dalexsoto) [inline src/chip.cs:1046]: There are other `ID` that needs to be `Id`

```suggestion
		void DisableNetwork (NSData networkId, ulong breadcrumb, uint timeoutMs, ChipResponseHandler responseHandler);
```

File: src/chip.cs [hunk 35]
```diff
@@ -0,0 +1,1767 @@
...
+		bool IsRunning { get; }
+
+		[Export ("pairDevice:discriminator:setupPINCode:error:")]
+		bool PairDevice (ulong deviceID, ushort discriminator, uint setupPinCode, [NullAllowed] out NSError error);
+
+		[Export ("pairDevice:address:port:discriminator:setupPINCode:error:")]
+		bool PairDevice (ulong deviceID, string address, ushort port, ushort discriminator, uint setupPinCode, [NullAllowed] out NSError error);
+
+		[Export ("pairDeviceWithoutSecurity:address:port:error:")]
+		bool PairDeviceWithoutSecurity (ulong deviceID, string address, ushort port, [NullAllowed] out NSError error);
```
Comment (dalexsoto) [inline src/chip.cs:1617]: Same search and replace. 

```suggestion
		bool PairDeviceWithoutSecurity (ulong deviceId, string address, ushort port, [NullAllowed] out NSError error);
```

File: src/chip.cs [hunk 36]
```diff
@@ -0,0 +1,1767 @@
...
+		NSNumber ControllerNodeId { get; }
+
+		[Export ("setPairingDelegate:queue:")]
+		void SetPairingDelegate (IChipDevicePairingDelegate @delegate, DispatchQueue queue);
+
+		[Export ("startup:")]
+		bool Startup ([NullAllowed] IChipPersistentStorageDelegate storageDelegate);
+
+		[Export ("shutdown")]
+		bool Shutdown();
```
Comment (dalexsoto) [inline src/chip.cs:1652]:
```suggestion
		bool Shutdown ();
```

File: src/chip.cs [hunk 37]
```diff
@@ -0,0 +1,1767 @@
...
+	interface ChipError
+	{
+		[Static]
+		[Export ("errorForCHIPErrorCode:")]
+		[return: NullAllowed]
+		NSError Create (int errorCode);
+
+		[Static]
+		[Export ("errorToCHIPErrorCode:")]
+		int ToChipErrorCode (NSError errorCode);
```
Comment (dalexsoto) [inline src/chip.cs:1667]:
```suggestion
		int ConvertToChipErrorCode (NSError errorCode);
```

**Review by dalexsoto — APPROVED** [review]
After the comments are addressed 👍

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tools/common/Target.cs [hunk 38]
```diff
@@ -222,6 +222,7 @@ public void GatherFrameworks ()
 						case "Metal":
 						case "MetalKit":
 						case "MetalPerformanceShaders":
+						case "CHIP":
```
Comment (rolfbjarne) [inline tools/common/Target.cs:225]: @dalexsoto out of curiosity, why doesn't the default case handle this?

https://github.com/xamarin/xamarin-macios/blob/401befe28f90698ba5306b7fd320a2bddf9001b9/tools/common/Target.cs#L251-L258

Comment (dalexsoto) [reply tools/common/Target.cs:225]: @rolfbjarne good question, I did not have a close look at it I wonder if there is a casing issue somewhere @mandel-macaque looks like we'll need to check deeper

Comment (mandel-macaque) [reply tools/common/Target.cs:225]: I'm looking atm.

**Pull request merged**